modules/ROOT/pages/patterns/variable-length-patterns.adoc

```diff
@@ -0,0 +1,544 @@
+
+[source, role=noheader]
+----
+(:Station { name: 'Denmark Hill' })<-[:CALLS_AT]-(:Stop)
+  -[:NEXT]->(:Stop)
+  -[:NEXT]->(:Stop)
+  -[:NEXT]->(:Stop)-[:CALLS_AT]->
+(:Station { name: 'Clapham Junction' })
+----
+
+[source, role=noheader]
+----
+(:Station { name: 'Denmark Hill' })<-[:CALLS_AT]-(:Stop)
+  -[:NEXT]->(:Stop)-[:CALLS_AT]->
+(:Station { name: 'Clapham Junction' })
+----
+
+To return both solutions in the same query using these fixed-length path patterns, a xref:clauses/union.adoc[UNION] of two `MATCH` statements would be needed.
+For example, the following query returns the `departure` of the two services:
+
+.Query
+[source, cypher]
+----
+MATCH (:Station { name: 'Denmark Hill' })<-[:CALLS_AT]-(d:Stop)
+        -[:NEXT]->(:Stop)
+        -[:NEXT]->(:Stop)
+        -[:NEXT]->(a:Stop)-[:CALLS_AT]->
+      (:Station { name: 'Clapham Junction' })
+RETURN d.departs AS departureTime, a.arrives AS arrivalTime
+UNION
+MATCH (:Station { name: 'Denmark Hill' })<-[:CALLS_AT]-(d:Stop)
+        -[:NEXT]->(a:Stop)-[:CALLS_AT]->
+      (:Station { name: 'Clapham Junction' })
+RETURN d.departs AS departureTime, a.arrives AS arrivalTime
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="2*<m"]
+|===
+
+| departureTime | arrivalTime
+
+| "17:07:00Z" | "17:19:00Z"
+| "17:10:00Z" | "17:17:00Z"
+
+2+d|Rows: 2
+
+|===
+
+The problem with this solution is that not only is it verbose, it can only be used where the lengths of the target paths are known in advance.
+Quantified path patterns solve this problem by extracting repeating parts of a path pattern into parentheses and applying a *quantifier*.
+That quantifier specifies a range of possible repetitions of the extracted pattern to match on.
+For the current example, the first step is identifying the repeating pattern, which in this case is the sequence of alternating `Stop` nodes and `NEXT` relationships, representing one segment of a `Service`:
+
+[source, role=noheader]
+----
+(:Stop)-[:NEXT]->(:Stop)
+----
+
+The shortest path has one instance of this pattern, the longest three.
+So the quantifier applied to the wrapper parentheses is the range one to three, expressed as `{1,3}`:
+
+[source, role=noheader]
+----
+((:Stop)-[:NEXT]->(:Stop)){1,3}
+----
+
+This also includes repetitions of two, but in this case this repetition will not return matches.
+To understand the semantics of this pattern, it helps to work through the expansion of the repetitions.
+Here are the three repetitions specified by the quantifier, combined into a union of path patterns:
+
+[source, role=noheader]
+----
+(:Stop)-[:NEXT]->(:Stop) |
+(:Stop)-[:NEXT]->(:Stop)(:Stop)-[:NEXT]->(:Stop) |
+(:Stop)-[:NEXT]->(:Stop)(:Stop)-[:NEXT]->(:Stop)(:Stop)-[:NEXT]->(:Stop)
+----
+
+The union operator (`|`) here is used for illustration only; using it this way is not part of Cypher syntax.
```

Comment:
This suggests that juxtaposing path patterns like that is part of Cypher:
```
(:Stop)-[:NEXT]->(:Stop)(:Stop)-[:NEXT]->(:Stop)
```

Reply:
Added a few words to make it clear that juxtaposing node patterns explicitly is not part of the syntax.